### PR TITLE
[TH2-1150] Add message router for message group batches

### PR DESCRIPTION
**Pull request by ivandruzhinin**
* make parsed and raw message routers use message group batches under the hood
 * remove some usages of Lombok's annotations because it makes annotated classes harder to use from Kotlin code

### CODE REVIEW

File: build.gradle
Comment (OptimumOpium) [inline build.gradle:130]: I think we should remove a wildcard for the maintenance version

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java:39]: We can use `@RequiredArgsConstructor` and `@Getter` annotation for class instead

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java:26]: I thing these fields might be final

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/parsed/RabbitParsedBatchSender.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/parsed/RabbitParsedBatchSender.java:59]: Why don't use `var` instead? We compile the project for java 11 so this feature is available. I don't think that we should use the Lombok in this case

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitBatchSubscriber.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitBatchSubscriber.java:84]: Fields can be final

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/raw/RabbitRawBatchSender.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/raw/RabbitRawBatchSender.java:51]: I think we should use Java `var` instead of Lombok `val`

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchQueue.kt
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchQueue.kt:1]: Please, add a copyright

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchRouter.kt
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchRouter.kt:1]: Please, add a copyright

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchSender.kt
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchSender.kt:1]: Please, add a copyright

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchSubscriber.kt
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/RabbitMessageGroupBatchSubscriber.kt:1]: Please, add a copyright

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitBatchSubscriber.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitBatchSubscriber.java:86]: Maybe we should reorder the `messageType` and `sessionAlias` parameters in the constructor to avoid two String parameters next to each other.
```suggestion
        public Metadata(long sequence, String messageType, Direction direction, String sessionAlias) {
```

**Review by OptimumOpium — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java
Comment (ivandruzhinin) [reply src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java:39]: Getters generated by Lombok aren't visibile to Kotlin compiler unless Kotlin code is in a different module

Comment (OptimumOpium) [reply src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java:39]: Ok, I get it. Maybe we should convert that class to a Kotlin data class?

Comment (ivandruzhinin) [reply src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/SubscribeTarget.java:39]: Yeah, we probably should

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/parsed/RabbitParsedBatchSubscriber.java
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/parsed/RabbitParsedBatchSubscriber.java:81]: We don't add anything to the created message batch

**Review by OptimumOpium — CHANGES_REQUESTED** [review]
Please, correct the logic for the `ParsedBatchSsubscriber`

File: build.gradle
Comment (ivandruzhinin) [reply build.gradle:132]: Yeah, but, probably, in another task

**Review by OptimumOpium — APPROVED** [review]
Looks fine to me. But I think @Nikita-Smirnov-Exactpro should take a look as well before we merge the changes

**Pull request merged**
